### PR TITLE
When comparing scope annotations, take their values into account

### LINKED ISSUE #377: Support arguments for scope annotations
For each scope we need to define a new scope annotation, e.g.
```kotlin
import me.tatarka.inject.annotations.Scope

@Scope annotation class SingleInAppScope
@Scope annotation class SingleInLoggedInScope
@Scope annotation class SingleInRendererScope
...
```
Rather, I'd like to define a single annotation and use different arguments:
```kotlin
@Scope
annotation class SingleIn(val scope: KClass<*>)
```
Then I only need to define new class references and can reuse the same scope:
```kotlin
@SingleIn(AppScope::class)
@SingleIn(LoggedInScope::class)
...
```
The challenge is that more argument types such as ints, strings and enums must be supported for this.

Note that scope annotations using this mechanism today compile fine, but `@SingleIn(AppScope::class)` would be treated the same as `@SingleIn(LoggedInScope::class)`. The argument is ignored for comparisons. 

The same should be supported for qualifiers #253.

### PR DESCRIPTION
**Pull request by evant**
ex: @NamedScope("one") and @NamedScope("two") are now treated as distinct scopes.

Fixes #377

### CODE REVIEW

File: integration-tests/common/src/main/kotlin/me/tatarka/inject/test/Scope.kt [hunk 1]
```diff
@@ -236,3 +236,25 @@ abstract class SuperclassScopeFooComponent : BaseSuperclassScopeFooComponent {
...
+@Inject
+@ScopeWithArg("parent")
+class ScopedParentFoo
+
+@Inject
+@ScopeWithArg("child")
+class ScopedChildFoo
+
+@Component
+@ScopeWithArg("parent")
```
Comment (vRallev) [inline integration-tests/common/src/main/kotlin/me/tatarka/inject/test/Scope.kt:252]: Curious, `KClass` arguments are supported as well, right? I didn't see how you implement the `equals()` check.

Comment (evant) [reply integration-tests/common/src/main/kotlin/me/tatarka/inject/test/Scope.kt:252]: They should be yes, it's using the same logic that was already used for qualifiers

Comment (evant) [reply integration-tests/common/src/main/kotlin/me/tatarka/inject/test/Scope.kt:252]: If you are curious https://github.com/evant/kotlin-inject/blob/094d8ce2c1a70a1b29fc08be073052fcf3a76e4a/ast/ksp/src/main/kotlin/me/tatarka/kotlin/ast/KSUtil.kt#L60

Comment (vRallev) [reply integration-tests/common/src/main/kotlin/me/tatarka/inject/test/Scope.kt:252]: Thanks!

**Pull request merged**